### PR TITLE
Consolidate base directory for diagnostics (obs, mapping and mask files) into one config option

### PR DESCRIPTION
**Pull request by xylar**
This merge consolidates the base directory for diagnostics in each config file into one `baseDirectory` option in the new `[diagnostics]` section.  This means there is only one option that needs to be changed in order to point to the diagnostics directory on each machine instead of needing to set this directory in 4 or 5 different places.

The new `[diagnostics]` section includes the `baseDirectory` as well as the mapping and region-masks subdirectories, using the standard directory layout as on the E3SM diagnostics server.

The various `*Observations` sections (for ocean, sea ice and icebergs) have also been modified to reference this new standard base directory so users will need to change very little to set up a new config file.

Many tasks and quite a bit of documentation needed to be updated to handle these seemingly minor changes.

Example config file have also been updated with new diagnostics section.  Various other sections and options were removed as a result of the simplification.

A bug was fixed (for my convenience in debugging the code)  in the `PlotClimatologyMpasSubtask` where `decode()` was being called on `depthSlice` even though this is already a string (not an array of bytes) in python 3.  This bug shouldn't affect analysis performed with python 2.

### CODE REVIEW

**Comment by xylar** [comment]
@milenaveneziani, this came up when I was trying to explain config files to Karthik on his visit last week.  It became clear to me that things are unnecessarily complicated now that we have a standard directory structure and should be simplified.

Let me know if you have concerns.  It would be great if you could run a test or two but I know you're preparing for AGU just like I am...

**Comment by milenaveneziani** [comment]
@xylar, I just took a quick look. I wonder if it is a good idea to not allow the user to specify the various directories manually, if they want to.
Pinging @vanroekel as well to see if he has an opinion on this.

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta [hunk 1]
```diff
@@ -112,67 +118,3 @@ endYear = 9999
...
-[timeSeriesSeaIceAreaVol]
-## options related to plotting time series of sea ice area and volume
-
-# plot on polar plot
-polarPlot = False
-
-[regions]
-# Directory containing mask files for ocean basins and ice shelves
-regionMaskDirectory = /lus/theta-fs0/projects/ClimateEnergy_2/diagnostics/mpas_analysis/region_masks
-
```
Comment (milenaveneziani) [inline configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: why is the section below removed from this config file (same goes for other sections in other config files)?

**Comment by xylar** [comment]
> @xylar, I just took a quick look. I wonder if it is a good idea to not allow the user to specify the various directories manually, if they want to.

You can still specify the paths to any of the directories as absolute paths, so someone could specify them manually.  However, I think we want to discourage folks from having their own directory structure going forward so I don't know that this capability needs to be well advertised.

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta (hunk 1 above)
Comment (xylar) [reply configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: That's the whole point of this PR.  People shouldn't need to point to these directories manually now that the directory structure is standardized.

Are you thinking that people would need to make their own region masks and point to those directories?  I guess that's conceivable, particularly for non-standard grids.

My thought was that we want the example config files to be relatively simple.  I guess you prefer to keep them relatively verbose and to point out to users that they could point the `mappingSubdirectory` and/or `regionMaskSubdirectory` as an absolute path?

**Comment by milenaveneziani** [comment]
Well, why not? If the user needs to play with the directory structure for one reason or another, we should let them know they can.
We could just do it in the config.example file (commenting out the specific directory names and mentioning the option).

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta (hunk 1 above)
Comment (milenaveneziani) [reply configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: No, in this case I only meant the section below that has other (unrelated) things in it.
I would only keep config.example more verbose.

File: configs/alcf/config.20180410.A_WCYCL1950_HR.ne120_oRRS18v3_ICG.theta [hunk 2]
```diff
@@ -111,52 +117,3 @@ endYear = 10
...
-[timeSeriesSeaIceAreaVol]
-## options related to plotting time series of sea ice area and volume
-
-# plot on polar plot
-polarPlot = False
-
-[regions]
-# Directory containing mask files for ocean basins and ice shelves
-regionMaskDirectory = /lus/theta-fs0/projects/ClimateEnergy_2/diagnostics/mpas_analysis/region_masks
-
```
Comment (milenaveneziani) [inline configs/alcf/config.20180410.A_WCYCL1950_HR.ne120_oRRS18v3_ICG.theta]: here is another example of section (below this comment) that seems unrelated to this PR and that is being removed. Not sure if it was intentional or not.

File: configs/anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil [hunk 3]
```diff
@@ -107,31 +124,6 @@ endYear = 22
...
-# directory where ocean observations are stored
-baseDirectory = /lcrc/group/acme/diagnostics/observations/Ocean
-
-[seaIceObservations]
-## options related to sea ice observations with which the results will be
-## compared
-
-# directory where sea ice observations are stored
-baseDirectory = /lcrc/group/acme/diagnostics/observations/SeaIce
-
```
Comment (milenaveneziani) [inline configs/anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil]: also this piece below about `timeSeriesSeaIceAreaVol`. And similarly for other config example files.

Comment (milenaveneziani) [reply configs/anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil]: Actually, it seems this may be the last config file with this issue.

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta (hunk 1 above)
Comment (xylar) [reply configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: I see.  The way comments work on GitHub, all I see are the previous 3 lines to your comment so that wasn't clear.   It could be that i need to put the MOC section back in but I don't think the rest of this is helpful.  It just makes it a lot harder to change default options because all of the example config files have to be changed to match.  I think the examples should be relatively lean.

**Comment by xylar** [comment]
> Well, why not? If the user needs to play with the directory structure for one reason or another, we should let them know they can.
We could just do it in the config.example file (commenting out the specific directory names and mentioning the option).

Okay, I think that sounds like a good plan.

File: configs/alcf/config.20180410.A_WCYCL1950_HR.ne120_oRRS18v3_ICG.theta (hunk 2 above)
Comment (xylar) [reply configs/alcf/config.20180410.A_WCYCL1950_HR.ne120_oRRS18v3_ICG.theta]: I really think we don't want this stuff in the example config files, so, yes, it was intentional.

File: configs/anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil (hunk 3 above)
Comment (xylar) [reply configs/anvil/config.20170926.FCT2.A_WCYCL1850S.ne30_oECv3.anvil]: These are all default options being copied into example config files.  I realize they might demonstrate to users some of the capabilities that they could customize but we have several hundred such options at this point and these few are just arbitrary in my view.  As I said, they make changing defaults very difficult and I think they confuse and overwhelm new users rather than providing helpful direction.  That's why I think we want to keep the examples as more the minimum set of options needed to analyze that particular example run and rely on documentation, config.example and config.default to point out additional options.

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta (hunk 1 above)
Comment (milenaveneziani) [reply configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: ok, agree that only the MOC section should be put back in for this specific config files, while the others are redundant.

File: configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta [hunk 4]
```diff
@@ -112,67 +118,3 @@ endYear = 9999
...
-## options related to plotting the streamfunction of the meridional overturning
-## circulation (MOC)
-
-# Use postprocessing script to compute the MOC? You want this to be True
-# for low-resolution simulations that use GM to parameterize eddies, because
-# the online MOC analysis member currently does not include the bolus velocity
-# in its calculation, whereas the postprocessing script does.
-# NOTE: this is a temporary option that will be removed once the online
-# MOC takes into account the bolus velocity when GM is on.
-usePostprocessingScript = True
```
Comment (xylar) [inline configs/alcf/config.20171031.tenYearTest.GMPAS-IAF.T62_oEC60to30v3wLI.60layer.theta]: I will put the MOC section back.  That was removed by mistake.

**Comment by xylar** [comment]
I believe I've addressed the concerns you raised but please let me know if you notice anything more that I missed.  I will re-test on my side but I would also appreciate it if you could do a test or 2 yourself, perhaps giving an absolute path to the regions, mapping or observations subdirectories and make sure it does, in fact, work for you.

**Comment by milenaveneziani** [comment]
No problem. Remember, I am a 'stuff keeper' :D

Ok, I will run a test hopefully tomorrow.

**Comment by xylar** [comment]
Ah, right!  I should cc Gennaro on these messages.  He would understand ;-)

**Comment by xylar** [comment]
@milenaveneziani, have you had a chance to test this one?  Are you happy with it now?

**Comment by xylar** [comment]
@milenaveneziani, I believe this PR is just waiting on testing from you. Is that correct?

**Comment by milenaveneziani** [comment]
oh cute, there is an actual stickler-ci reviewer! funny.
yes, thanks for the reminder @xylar, I'll test this shortly.

**Comment by xylar** [comment]
Excellent!  Thank you for testing, @milenaveneziani.
